randovania/game_description/db/dock_node.py

```diff
@@ -118,3 +118,6 @@ def connections_from(self, context: NodeContext) -> typing.Iterator[tuple[Node,
 
         if (lock_connection := self._lock_connection(context)) is not None:
             yield lock_connection
+
+    def is_teleporter_node(self) -> bool:
+        return self.dock_type.short_name in {"teleporter", "elevator", "shuttle"}
```

Comment:
What would be a good way to replace these `is_teleporter_node` all over the place? It basically just replaced `isinstance(node, TeleporterNode)` atm.

Reply:
Looking at what is still left there: I still don't know if it maybe should be just handled via `DockWeaknessDatabase` by setting something like `is_teleporther_thing` and `ignore_for_hints` in `DockType`. 
That would remove every hard coded check for `teleporter`, `elevator` or `portal` strings...hopefully.